packages/presentation/lib/src/pages/widgets/game_field.dart

```diff
@@ -1,54 +1,75 @@
+import 'package:domain/domain.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:presentation/src/bloc/game_bloc.dart';
-import 'package:presentation/src/bloc/state.dart';
+import 'package:presentation/src/game_bloc/bloc_screen.dart';
+import 'package:presentation/src/game_bloc/bloc_tile.dart';
+import 'package:presentation/src/game_bloc/game_bloc.dart';
 import 'package:presentation/src/pages/widgets/game_form.dart';
 
-const attemptsCount = 3;
-
-class GameField extends StatelessWidget {
+class GameField extends BlocScreen {
   const GameField({Key? key}) : super(key: key);
 
-  void _showEndGameMessage(BuildContext context, String message) {
-    ScaffoldMessenger.of(context).showSnackBar(
-      SnackBar(
-        content: Text(message),
-        duration: const Duration(seconds: 1),
-      ),
+  @override
+  State<GameField> createState() => _GameFieldState();
+}
+
+class _GameFieldState extends BlocScreenState<GameField, GameBloc> {
+  _GameFieldState()
+      : super(
+          GameBloc(
+            GenerateGuessNumberUseCase(),
+            MakeAttemptUseCase(),
+          ),
+        );
+
+  void _showEndGameMessage(String message) {
+    WidgetsBinding.instance.addPostFrameCallback(
+      (_) {
+        ScaffoldMessenger.of(context).showSnackBar(
+          SnackBar(
+            content: Text(message),
+            duration: const Duration(seconds: 1),
+          ),
+        );
+      },
     );
   }
 
-  void _showAttemptsRemain(BuildContext context, int guessAttemptsCount) {
-    final message = guessAttemptsCount == attemptsCount
-        ? 'Game started!'
-        : 'Attempts left: $guessAttemptsCount';
-    ScaffoldMessenger.of(context).showSnackBar(
-      SnackBar(
-        content: Text(message),
-        duration: const Duration(seconds: 1),
-      ),
+  void _showAttemptsRemain(int guessAttemptsCount, BlocTileState state) {
+    WidgetsBinding.instance.addPostFrameCallback(
+      (_) {
+        final message = state == BlocTileState.gameInProgress
+            ? 'Game started!'
+            : 'Attempts left: $guessAttemptsCount';
+        ScaffoldMessenger.of(context).showSnackBar(
+          SnackBar(
+            content: Text(message),
+            duration: const Duration(seconds: 1),
+          ),
+        );
+      },
     );
```

Comment:
better to handle such events via separate stream. u can see how Katya or Liza implemented it